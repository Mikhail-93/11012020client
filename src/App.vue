<template>
    <v-app id="inspire">
        <v-navigation-drawer v-model="drawer" app>
            <v-list dense>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-home</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Home</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-contact-mail</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Contact</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
            </v-list>
        </v-navigation-drawer>

        <v-app-bar
                app
                color="indigo"
                dark
        >
            <v-app-bar-nav-icon @click.stop="drawer = !drawer"/>
            <v-toolbar-title>Application</v-toolbar-title>
        </v-app-bar>

        <v-content>
            <v-container class="fill-height pa-0" fluid>
                <LMap ref="map" style="z-index: 0">
                    <l-tile-layer
                            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                            attribution='Map data: &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)'
                    >
                    </l-tile-layer>
                    <!--                    <l-control-attribution position="bottomright" prefix="leaflet OpenStreetMap Contributions" >-->

                </LMap>
                <!--                class="align-stretch"-->
            </v-container>
        </v-content>
    </v-app>
</template>

<script>
    import {LMap, LTileLayer, LMarker, LControlAttribution} from 'vue2-leaflet';

    export default {
        components: {
            LMap,
            LTileLayer,
            LMarker,
            LControlAttribution
        },
        props: {
            source: String,
        },
        data: () => ({
            drawer: null,
        }),
        mounted() {
            this.$nextTick(() => {
                // this.map=
                this.$refs.map.mapObject.setView([51.505, -0.09], 13)

            })
        }
    }
</script>