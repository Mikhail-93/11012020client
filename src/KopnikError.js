class KopnikError extends Error {
    constructor(message, code) {
        super(message)
        this.code = code
    }
}
class KopnikApiError extends KopnikError {
    constructor(message, code, url) {
        super(message, code)
        this.url = url
    }
}



export {KopnikError, KopnikApiError}
