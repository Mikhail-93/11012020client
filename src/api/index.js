import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
