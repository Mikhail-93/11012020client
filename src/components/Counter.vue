<template>
    <div>
        <span class="count">{{ count.count }}</span>
        <button @click="increment">Increment</button>
    </div>
</template>
<script>
    export default {
        data() {
            return {
                // count:{
                //     count: 0
                // }
            }
        },
        props:{
            count:{
                type: Object,
            }
        },

        methods: {
            increment() {
                this.count.count++
            }
        }
    }
</script>
