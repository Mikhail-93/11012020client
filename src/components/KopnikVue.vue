<template>
    <v-list v-if="value.isLoaded">
        <v-list-item>
            <avatar-vue :value="value" :size="avatarSize" class="{avatarMxAuto: 'mx-auto'}"
                          @click="avatar_click" @dblclick="avatar_dblclick">
            </avatar-vue>
        </v-list-item>
        <v-list-item v-if="!fio">
            <v-list-item-content>
                <v-list-item-title class="title">{{ value.name }}</v-list-item-title>
            </v-list-item-content>
        </v-list-item>
        <v-list-item v-if="locale">
            <v-list-item-content>
                <v-combobox ref="locale"
                            :readonly="readonly"
                            :return-object="false"
                            :allow-overflow="false"
                            v-model="value.locale"
                            :items="locales"
                            :auto-select-first="true"
                            label="Язык / Language"
                            @change="$emit('locale_change', $event)"
                >
                </v-combobox>
            </v-list-item-content>
        </v-list-item>

        <v-list-item v-if="fio">
            <v-list-item-content>
                <ValidationProvider name="lastName" rules="required" v-slot="{ errors, valid }">
                    <v-text-field
                            v-model="value.lastName"
                            :label="$t('profile.lastName')"
                            :error-messages="errors"
                            :success="valid"
                            :readonly="readonly"
                    >
                    </v-text-field>
                </ValidationProvider>
            </v-list-item-content>
        </v-list-item>
        <v-list-item v-if="fio">
            <v-list-item-content>
                <ValidationProvider name="firstName" rules="required" v-slot="{ errors, valid }">
                    <v-text-field
                            v-model="value.firstName"
                            :label="$t('profile.firstName')"
                            :error-messages="errors"
                            :success="valid"
                            :readonly="readonly"
                    ></v-text-field>
                </ValidationProvider>
            </v-list-item-content>
        </v-list-item>
        <v-list-item v-if="fio">
            <v-list-item-content>
                <ValidationProvider name="patronymic" rules="required" v-slot="{ errors, valid }">
                    <v-text-field
                            v-model="value.patronymic"
                            :label="$t('profile.patronymic')"
                            :error-messages="errors"
                            :success="valid"
                            :readonly="readonly"
                    ></v-text-field>
                </ValidationProvider>
            </v-list-item-content>
        </v-list-item>
        <v-list-item v-if="role">
            <v-list-item-content>
                <ValidationProvider name="role" rules="required" v-slot="{ errors, valid }">
                    <v-radio-group v-model="value.role">
                        <template v-slot:label>
                            <div>{{ $t('profile.role.title')}}</div>
                        </template>
                        <v-radio :value="1" :label="$t('profile.role.trueKopnik.title')">
                            <template v-slot:label>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.trueKopnik.criteria[0]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.trueKopnik.criteria[1]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.trueKopnik.criteria[2]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.trueKopnik.criteria[3]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.trueKopnik.criteria[4]') }}</v-checkbox>
                            </template>
                        </v-radio>
                        <v-radio :value="2" :label="$t('profile.role.danilovKopnik.title')">
                            <template v-slot:label>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.danilovKopnik.criteria[0]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.danilovKopnik.criteria[1]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.danilovKopnik.criteria[2]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.danilovKopnik.criteria[3]') }}</v-checkbox>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.danilovKopnik.criteria[4]') }}</v-checkbox>
                            </template>
                        </v-radio>
                        <v-radio :value="3" :label="$t('profile.role.futureKopnik.title')">
                            <template v-slot:label>
                                <v-checkbox value="true" readonly color="success">{{ $t('profile.role.futureKopnik.criteria[0]') }}</v-checkbox>
                            </template>
                        </v-radio>
                        <v-radio :value="4" :label="$t('profile.role.women.title')">
                        </v-radio>
                    </v-radio-group>
                </ValidationProvider>
            </v-list-item-content>
        </v-list-item>

        <v-list-item v-if="birthyear">
            <v-list-item-content>
                <ValidationProvider name="birthyear" rules="required|numeric|length:4"
                                    v-slot="{ errors, valid }">
                    <v-text-field
                            v-model="value.birthyear"
                            :label="$t('profile.birthyear')"
                            :v-mask="['####']"
                            :error-messages="errors"
                            :success="valid"
                            :readonly="readonly"
                    ></v-text-field>
                </ValidationProvider>
            </v-list-item-content>
        </v-list-item>
        <v-list-item v-if="passport">
            <v-list-item-content>
                <ValidationProvider name="passport" rules="required|numeric|length:4"
                                    v-slot="{ errors, valid }">
                    <v-text-field
                            v-model="value.passport"
                            mask="####"
                            :label="$t('profile.passport')"
                            :error-messages="errors"
                            :success="valid"
                            :readonly="readonly"
                    ></v-text-field>
                </ValidationProvider>
            </v-list-item-content>
        </v-list-item>
        <v-list-item v-if="location">
            <v-list-item-content>
                <v-list-item-title  class="mb-3" style="white-space: inherit !important;">{{ $t('profile.location') }}</v-list-item-title>
                <MapVue :center="value.location" :zoom="14"
                        :zoom-control="true" :layers-control="false" :locate-control="true"
                        @update:center="$emit('map_updateCenter', $event)"
                        class="" style="z-index: 0; height: 50vh;">
                    <l-marker :lat-lng="value.location"></l-marker>
                </MapVue>
            </v-list-item-content>
        </v-list-item>
        <slot></slot>
    </v-list>
</template>
<script>
    import {LMarker} from 'vue2-leaflet'
    import Kopnik from "../models/Kopnik"
    import MapVue from "./MapVue";
    import AvatarVue from "./AvatarVue";
    import logger from "./mixin/logger"
    import {container} from "../bottle/bottle";
    import {
        ValidationProvider,
    } from "vee-validate"

    export default {
        name: "Kopnik",
        mixins: [logger],
        components: {
            MapVue,
            LMarker,
            ValidationProvider,
            AvatarVue,
        },
        data: () => {
            return {
                application: container.application,
                locales: [
                    {
                        text: "Русский",
                        value: "ru"
                    },
                    {
                        text: "English",
                        value: "en"
                    }
                ],
            }
        },
        props: {
            avatarSize: {
                type: Number,
                default: 128
            },
            avatarTile: {
                type: Boolean,
                default: false
            },
            avatarMxAuto: {
                type: Boolean,
                default: false
            },
            value: {
                type: Kopnik
            },
            locale: {
                type: Boolean,
                default: false,
            },
            fio: {
                type: Boolean,
                default: false,
            },
            birthyear: {
                type: Boolean,
                default: false
            },
            passport: {
                type: Boolean,
                default: false
            },
            to: {
                type: [String, Object]
            },
            location: {
                type: Boolean,
                default: false
            },
            readonly: {
                type: Boolean,
                default: false
            },
            role:{
                type: Boolean,
                default: false,
            }
        },
        filters: {
            undefined(value) {
                return value === undefined ? '' : value
            }
        },
        computed: {},
        watch: {},
        methods: {
            avatar_click() {
                this.$emit('click', this.value)
            },
            avatar_dblclick() {
                this.$emit('dblclick', this.value)
            },
        },
        async created() {
        },
        async mounted() {
            await this.value.loaded()
        }
    }
</script>
