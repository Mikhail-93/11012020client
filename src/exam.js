const cars = {};Debugger attached.
const carHonda = {model: 'CR-V', year: 2015};
const carVolvo = {model: 'XC-60', year: 2018};
const carBMW = {model: 'X5', year: 2019};
cars[carHonda] = 'Good';
cars[carVolvo] = 'Better';
cars[carBMW] = 'The best';
console.log('Honda is', cars[carHonda]);
console.log('Volvo is', cars[carVolvo]);
console.log('BMW is', cars[carBMW]);
