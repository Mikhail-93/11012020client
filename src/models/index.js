/**
 * Created by alexey2baranov on 8/20/16.
 */

import AbstractSync from "./AbstractSync" //для того чтобы правильно работали рекурсивные ссылки модулей
// import Registration from "./model/Registration"
import Kopnik from "./Kopnik"
// import Kopa from "./model/Kopa"
// import File from "./model/File"

export {AbstractSync, Kopnik}