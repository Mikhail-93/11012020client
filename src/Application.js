import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
