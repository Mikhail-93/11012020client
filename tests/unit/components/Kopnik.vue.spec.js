import 'isomorphic-fetch'
import '../../vue-setup'
import {mount} from '@vue/test-utils'
import Vue from 'vue'

import KopnikVue from '../../../src/components/KopnikVue'
import Counter from '../../../src/components/Counter'
import {Kopnik} from "../../../src/models";
import flushPromises from "flush-promises";

describe('Counter', () => {
    describe('sub', () => {
        const count = {count: 0}
        // Теперь монтируем компонент и получаем wrapper
        const wrapper = mount(Counter, {
            propsData: {
                count: count
            }
        })

        it('html', () => {
            expect(wrapper.html()).toContain('<span class="count">0</span>')
        })

        // также легко проверить наличие других элементов
        it('button', () => {
            expect(wrapper.contains('button')).toBe(true)
        })

        it('нажатие кнопки должно увеличивать счётчик', async () => {
            expect(wrapper.vm.count.count).toBe(0)
            expect(wrapper.text()).toContain('0')
            const button = wrapper.find('button')
            button.trigger('click')
            expect(wrapper.vm.count.count).toBe(1)
            count.count = 2
            expect(wrapper.vm.count.count).toBe(2)
            /*        wrapper.setProps({
                        count:{count:1}
                    })*/
            await Vue.nextTick()
            await flushPromises()
            expect(wrapper.text()).toContain('0')
        })
    })
})

describe('unit.components.Kopnik.vue', () => {
    it.skip('renders firstName', async () => {
        const kopnik2 = Kopnik.getReference(2)
        const wrapper = mount(KopnikVue, {
            propsData: {value: kopnik2}
        })
        // await kopnik2.loaded()
        kopnik2.firstName = 'Алексей'
        await Vue.nextTick()
        await Vue.nextTick()
        await Vue.nextTick()
        // console.log(wrapper.vm.$props.value.plain)
        expect(wrapper.text()).toMatch(/Алексей/)
    })
    it('renders', async () => {
        const kopnik2 = await Kopnik.get(2)
        const wrapper = mount(KopnikVue, {
            propsData: {
                value: kopnik2,
                passport: true,
                birthyear: true
            }
        })
        expect(wrapper.html()).toMatchSnapshot()//(/Баранов Алексей/)
    })
})

