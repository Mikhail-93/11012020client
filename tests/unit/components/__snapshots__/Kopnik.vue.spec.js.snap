// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit components Kopnik render 1`] = `
<div
  class="v-list v-sheet v-sheet--tile theme--light"
  role="list"
>
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <span
      class="v-badge {avatarMxAuto: 'mx-auto'} v-badge--bottom theme--light"
    >
      <div
        class="v-avatar"
        style="height: 128px; min-width: 128px; width: 128px;"
      >
        <img
          src="/avatar.png"
          style="object-fit: cover;"
        />
      </div>
      <span
        class="v-badge__wrapper"
      >
        <span
          aria-atomic="true"
          aria-label="знак"
          aria-live="polite"
          class="v-badge__badge orange"
          role="status"
        >
          1
        </span>
      </span>
    </span>
  </div>
   
  <!---->
   
  <!---->
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <span>
        <div
          class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light v-text-field v-text-field--is-booted success--text"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light success--text"
                  for="input-12"
                  style="left: 0px; position: absolute;"
                >
                  Фамилия
                </label>
                <input
                  id="input-12"
                  readonly="readonly"
                  required="required"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light success--text"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <span>
        <div
          class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light success--text"
                  for="input-17"
                  style="left: 0px; position: absolute;"
                >
                  Имя
                </label>
                <input
                  id="input-17"
                  required="required"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light success--text"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <span>
        <div
          class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light success--text"
                  for="input-22"
                  style="left: 0px; position: absolute;"
                >
                  Отчество
                </label>
                <input
                  id="input-22"
                  required="required"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light success--text"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <span>
        <div
          class="v-input v-input--has-state theme--light v-text-field v-text-field--is-booted success--text"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label theme--light success--text"
                  for="input-27"
                  style="left: 0px; position: absolute;"
                >
                  Прозвище
                </label>
                <input
                  id="input-27"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light success--text"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <span>
        <div
          class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light success--text"
                  for="input-32"
                  style="left: 0px; position: absolute;"
                >
                  Год рождения
                </label>
                <input
                  id="input-32"
                  required="required"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light success--text"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <span>
        <div
          class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light success--text"
                  for="input-37"
                  style="left: 0px; position: absolute;"
                >
                  Последние 4 цифры пасспорта
                </label>
                <input
                  id="input-37"
                  required="required"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light success--text"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
              <div
                class="v-counter theme--light"
              >
                4 / 4
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <div
        class="vue2leaflet-map leaflet-container leaflet-grab leaflet-touch-drag"
        style="z-index: 0; height: 50vh; position: relative;"
        tabindex="0"
      >
        <div />
        <div />
        <div />
         
        <!---->
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; left: 0px; top: 0px;"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://c.tile.openstreetmap.org/14/14925/5759.png"
                  style="width: 256px; height: 256px; left: -27px; top: -213px;"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-shadow-pane"
          >
            <img
              alt=""
              class="leaflet-marker-shadow leaflet-zoom-hide"
              src="marker-shadow.png"
              style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 0px; top: 0px;"
            />
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="marker-icon.png"
              style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 0px; top: 0px; z-index: 0;"
              tabindex="0"
            />
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          />
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JS library for interactive maps"
              >
                Leaflet
              </a>
               | Map data: © 
              <a
                href="http://www.openstreetmap.org/copyright"
              >
                OpenStreetMap
              </a>
              , 
              <a
                href="http://viewfinderpanoramas.org"
              >
                SRTM
              </a>
               | Map style: © 
              <a
                href="https://opentopomap.org"
              >
                OpenTopoMap
              </a>
               (
              <a
                href="https://creativecommons.org/licenses/by-sa/3.0/"
              >
                CC-BY-SA
              </a>
              )
            </div>
          </div>
        </div>
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
      </div>
    </div>
  </div>
   
</div>
`;

exports[`unit components Kopnik render short 1`] = `
<div
  class="v-list v-sheet v-sheet--tile theme--light"
  role="list"
>
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <span
      class="v-badge {avatarMxAuto: 'mx-auto'} v-badge--bottom theme--light"
    >
      <div
        class="v-avatar"
        style="height: 128px; min-width: 128px; width: 128px;"
      >
        <img
          src="/avatar.png"
          style="object-fit: cover;"
        />
      </div>
      <span
        class="v-badge__wrapper"
      >
        <span
          aria-atomic="true"
          aria-label="знак"
          aria-live="polite"
          class="v-badge__badge orange"
          role="status"
        >
          1
        </span>
      </span>
    </span>
  </div>
   
  <div
    class="v-list-item theme--light"
    role="listitem"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <div
        class="v-list-item__title title"
      >
        Сысоев Павел Якушев
      </div>
    </div>
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
   
</div>
`;
